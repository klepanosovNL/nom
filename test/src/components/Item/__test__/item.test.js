import React from 'react';
import renderer from 'react-test-renderer';
import * as reactRedux from 'react-redux';
import { render, screen, fireEvent, waitFor } from '@testing-library/react';
import { Item } from '../Item';
import { mockStore, wrapProvider } from '../../../utils/mockedStore';
// import { checkBoxToggle } from '../../../store/actionCreators';
import { toggleCheckboxStatus } from '../../../utils/utils';

jest.mock('../../../store/actionCreators', () => ({
	checkBoxToggle: (...args) => ({ type: 'TEST_TYPE', args }),
}));

describe('Item"s tests', () => {
	beforeEach(() => {
		useSelectorMock.mockClear();
		useDispatchMock.mockClear();
	});

	const useSelectorMock = jest.spyOn(reactRedux, 'useSelector');
	const useDispatchMock = jest.spyOn(reactRedux, 'useDispatch');

	it('Sent only name', () => {
		const tree = renderer.create(<Item name='name1'></Item>).toJSON();
		expect(tree).toMatchSnapshot();
	});

	it('Sent only description', () => {
		const tree = renderer
			.create(<Item description='description'></Item>)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});

	it('Sent only isDisabled=true', () => {
		const tree = renderer.create(<Item isDisabled='true'></Item>).toJSON();
		expect(tree).toMatchSnapshot();
	});

	it('Sent only isDisabled=false', () => {
		const tree = renderer.create(<Item></Item>).toJSON();
		expect(tree).toMatchSnapshot();
	});

	it('Sent nothing', () => {
		const tree = renderer.create(<Item></Item>).toJSON();
		expect(tree).toMatchSnapshot();
	});

	it('Sent all', () => {
		const tree = renderer
			.create(
				<Item name='name1' description='description' isDisabled='true'></Item>
			)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});

	it('Check event: handleInputChange(name)', async () => {
		const initialState = {
			list: [
				{
					name: 'dog',
					description: 'description',
				},
			],
			currentPreset: 'none',
			filterByCategories: '',
			filterByNameInput: '',
			blockedButton: '',
			customs: [],
		};

		const store = mockStore(initialState);

		render(
			wrapProvider(<Item name='dog' description='description'></Item>, store)
		);

		const label = screen.getByLabelText(/dog/i);
		fireEvent.click(label);

		await waitFor(() => {
			expect(store.getActions()).toEqual([
				{
					type: 'TEST_TYPE',
					args: [],
				},
			]);
		});
	});
});
