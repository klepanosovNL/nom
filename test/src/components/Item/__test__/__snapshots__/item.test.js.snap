// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Item"s tests Sent all 1`] = `
<div
  className="items__item"
>
  <label
    className="items__item__title"
  >
    <input
      onChange={[Function]}
      type="checkbox"
    />
    <i
      className="fa fa-light fa-ban"
    />
    name1
  </label>
  <div
    className="items__item__container"
  >
    <i
      className="fa fa-exclamation-circle"
    />
    <div
      className="items__item__description"
    >
      description
    </div>
  </div>
</div>
`;

exports[`Item"s tests Sent nothing 1`] = `
<div
  className="items__item"
>
  <label
    className="items__item__title"
  >
    <input
      onChange={[Function]}
      type="checkbox"
    />
    <i
      className="fa fa-duotone fa-check"
    />
  </label>
  <div
    className="items__item__container"
  >
    <i
      className="fa fa-exclamation-circle"
    />
    <div
      className="items__item__description"
    />
  </div>
</div>
`;

exports[`Item"s tests Sent only description 1`] = `
<div
  className="items__item"
>
  <label
    className="items__item__title"
  >
    <input
      onChange={[Function]}
      type="checkbox"
    />
    <i
      className="fa fa-duotone fa-check"
    />
  </label>
  <div
    className="items__item__container"
  >
    <i
      className="fa fa-exclamation-circle"
    />
    <div
      className="items__item__description"
    >
      description
    </div>
  </div>
</div>
`;

exports[`Item"s tests Sent only isDisabled=false 1`] = `
<div
  className="items__item"
>
  <label
    className="items__item__title"
  >
    <input
      onChange={[Function]}
      type="checkbox"
    />
    <i
      className="fa fa-duotone fa-check"
    />
  </label>
  <div
    className="items__item__container"
  >
    <i
      className="fa fa-exclamation-circle"
    />
    <div
      className="items__item__description"
    />
  </div>
</div>
`;

exports[`Item"s tests Sent only isDisabled=true 1`] = `
<div
  className="items__item"
>
  <label
    className="items__item__title"
  >
    <input
      onChange={[Function]}
      type="checkbox"
    />
    <i
      className="fa fa-light fa-ban"
    />
  </label>
  <div
    className="items__item__container"
  >
    <i
      className="fa fa-exclamation-circle"
    />
    <div
      className="items__item__description"
    />
  </div>
</div>
`;

exports[`Item"s tests Sent only name 1`] = `
<div
  className="items__item"
>
  <label
    className="items__item__title"
  >
    <input
      onChange={[Function]}
      type="checkbox"
    />
    <i
      className="fa fa-duotone fa-check"
    />
    name1
  </label>
  <div
    className="items__item__container"
  >
    <i
      className="fa fa-exclamation-circle"
    />
    <div
      className="items__item__description"
    />
  </div>
</div>
`;
